components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2184,40 +2182,18 @@ where
         // It's not correct anymore, so set it to None to let split checker update it.
         self.fsm.peer.approximate_size = None;
         self.fsm.peer.approximate_keys = None;
-
-        let is_leader = self.fsm.peer.is_leader();
-        let mut no_tick = false;
-        if is_leader {
-            self.fsm.peer.approximate_size = estimated_size;
-            self.fsm.peer.approximate_keys = estimated_keys;
-
-            if let Some(estimated_size) = self.fsm.peer.approximate_size {
-                if estimated_size > self.ctx.coprocessor_host.cfg.region_max_size.0 {
-                    info!(
-                        "trigger split check immediately";
-                        "region_id" => self.fsm.region_id(),
-                        "peer_id" => self.fsm.peer_id(),
-                        "size" => estimated_size,
-                    );
-                    self.fsm.peer.approximate_size = None;
-                    // trigger a split check immediately when size is still large
-                    self.ctx
-                        .router
-                        .force_send(region_id, PeerMsg::Tick(PeerTicks::SPLIT_REGION_CHECK))
-                        .unwrap();
-                    no_tick = true;
-                }
-            }
-        }
-        if !no_tick {
-            self.register_split_region_check_tick();
-        }
-
         let mut meta = self.ctx.store_meta.lock().unwrap();
         meta.set_region(&self.ctx.coprocessor_host, derived, &mut self.fsm.peer);
         self.fsm.peer.post_split();
 
+        // If `estimated_size` or `estimated_keys` equals none, the leader will schedule a
+        // split-check task in
+        let need_check_split = estimated_size.is_some() && estimated_keys.is_some();
```

Comment:
Maybe `estimated_size.is_none() || estimated_keys.is_none()` ? (To keep consistent with the above comment)

Reply:
No. If `estimated_size ` is none, the leader will schedule a split-check task in method `heartbeat_pd`.

Reply:
I see. Maybe we should make the comment more clear? 😂